### PR TITLE
Code gen

### PR DESCRIPTION
**Pull request by NDobrev**
# What

> Initial code generator

# Who

> Marto - 25192
Check

- Do not commit binary files
- Do not commit generated project files

### CODE REVIEW

File: JSImpl/src/ByteCodeGenerator.cpp [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+		double Number;
+	};
+	IPLVector<IPLString> m_RegisterTable;
+	IPLVector<Instruction> m_Code;
+
+	IPLStack<IPLString> m_RegisterStack;
+	unsigned m_CurrentStackPointer;

	IPLString m_OutputCode;
};
```
Comment (googleson78) [inline JSImpl/src/ByteCodeGenerator.cpp]: This is a coding style guide violation.

Comment (NDobrev) [reply JSImpl/src/ByteCodeGenerator.cpp]: something weird is happening between my editor and line ending in this file

Comment (NDobrev) [reply JSImpl/src/ByteCodeGenerator.cpp]: but anyway it's outdated :D

Comment (googleson78) [reply JSImpl/src/ByteCodeGenerator.cpp]: Not sure why I was seeing the outdated version.

**Pull request merged**